### DISCUSSION TITLE
ability to reset sortby not only on data change

### DISCUSSION THREAD

**Discussion by wwsskkaa** [TanStack/table#2685] [General]
im seeing resetResizing option to be able to reset the resizing by, for example, a user click. (official example: https://react-table.tanstack.com/docs/examples/column-resizing)

but for the useSortBy hook, im only seeing this option: autoResetSortBy: Boolean which the sortBy state will automatically reset if data is changed. 

i am trying to see if its doable for me to have a button just say reset sort, and it just automatically reset to initial state if user clicks it (no data is changed)

**Comment by gargroh** [comment]
you can call `setSortBy` method with `[]`

**Reply by wwsskkaa** [reply]
is there any documentation on setSortBy? I can't find it anywhere in the website https://react-table.tanstack.com/docs/api/useSortBy
also it would be very nice to provide an example, do I call this in a useEffect on a table? and do I get this from useTable instance?

**Reply by gargroh** [reply]
Feel free to PR for docs and example, well i mean its open source if you find its not perfect then you can contribute :)

You can use setSortBy in same way as toggleSortBy is used.

**Reply by wwsskkaa** [reply]
I see, but what is the difference between setSortBy and toggleSortBy then? to my understanding, if I want the table to sort by column firstName ascending order, then i can either call setSortBy([{id: firstName, desc: false}]) but I can also call toggleSortBy('firstName', false, false), so why do we need two methods here?

**Comment by gargroh** [comment]
setSortBy cam help you with multiple columns at once.

**Reply by wwsskkaa** [reply]
I see, so if im just aiming for single sort, then i can totally just use toggleSortBy there.
